core/src/main/scala/knobs/MutableConfig.scala

```diff
@@ -142,30 +142,38 @@ case class MutableConfig(root: String, base: BaseConfig) {
    * Fetch the `Map` that maps names to values. Turns the config into a pure
    * value disconnected from the file resources it came from.
    */
-  def getEnv: Task[Env] = immutable.map(_.env)
+  def getEnv: IO[Env] = immutable.map(_.env)
 
   /**
    * Get an immutable `Config` from of the current state of this
    * `MutableConfig`.
    */
-  def immutable: Task[Config] = base at root
+  def immutable: IO[Config] = base at root
 
   /**
    * Subscribe to notifications. The given handler will be invoked when any
    * change occurs to a configuration property that matches the pattern.
    */
-  def subscribe(p: Pattern, h: ChangeHandler): Task[Unit] =
-    base.subs.modify(_.insertWith(p local root, List(h))(_ ++ _))
-
-  import scalaz.stream.Process
+  def subscribe(p: Pattern, h: ChangeHandler): IO[Unit] =
+    base.subs.modify { map =>
+      map.get(p.local(root)) match {
+        case None           => map + ((p.local(root), List(h)))
+        case Some(existing) => map + ((p.local(root), existing ++ List(h)))
+      }
+    }
 
   /**
    * A process that produces chages to the configuration properties that match
    * the given pattern
    */
-  def changes(p: Pattern): Process[Task, (Name, Option[CfgValue])] = {
-    import scalaz.stream.async.signalOf
-    val sig = signalOf[(Name, Option[CfgValue])](("",None)) // TP: Not sure about the soundness of this default?
-    Process.eval(subscribe(p, (k, v) => sig.set((k, v)))).flatMap(_ => sig.discrete).drop(1) // droping the previously initilized tuple of the signal.
+  def changes(p: Pattern)(implicit ex: ExecutionContext): Stream[IO, (Name, Option[CfgValue])] = {
+    val signal = signalOf[IO, (Name, Option[CfgValue])](("", None)) // TP: Not sure about the soundness of this default?
+
+    Stream.eval {
+      for {
+        sig <- signal
+        _   <- subscribe(p, (k, v) => sig.set((k, v)))
```

Comment:
Creating a signal in FS2 gives an `IO[Signal[...]]` , had to do some dancing around to get this working.